### PR TITLE
Refactor to match other IoC extension's API

### PR DESCRIPTION
**Pull request by jeyjeyemem**

### CODE REVIEW

File: Src/Xer.Cqrs.Extensions.SimpleInjector/ICqrsEventHandlerSelector.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+using System.Reflection;
+using SimpleInjector;
+
+namespace Xer.Cqrs.Extensions.SimpleInjector
+{
+    public interface ICqrsEventHandlerSelector
+    {
+        ICqrsEventHandlerSelector ByInterface(Assembly assembly);
+        ICqrsEventHandlerSelector ByInterface(Assembly assembly, Lifestyle lifestyle);
+        ICqrsEventHandlerSelector ByInterface(IEnumerable<Assembly> assemblies);
```
Comment (mvput) [inline Src/Xer.Cqrs.Extensions.SimpleInjector/ICqrsEventHandlerSelector.cs]: Why not use the `params Assembly[] assemblies` syntax?

File: Src/Xer.Cqrs.Extensions.SimpleInjector/CqrsBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                    return new EventDelegator(CompositeMessageHandlerResolver.Compose(resolverArray));
+                }
+
+                return new EventDelegator(new SingleMessageHandlerRegistration().BuildMessageHandlerResolver());
+            });
+
+            return this;
+        }
+
+        private bool tryGetInstance<T>(out T instance) where T : class
```
Comment (mvput) [inline Src/Xer.Cqrs.Extensions.SimpleInjector/CqrsBuilder.cs]: Use the default guidelines `TryGetInstance`

File: Src/Xer.Cqrs.Extensions.SimpleInjector/ICqrsEventHandlerSelector.cs (hunk 1 above)
Comment (jeyjeyemem) [reply Src/Xer.Cqrs.Extensions.SimpleInjector/ICqrsEventHandlerSelector.cs]: I needed to pass in a Lifestyle as 2nd parameter to set lifestyle of handlers. I had some problems using params due to method overload resolution.

ByInterface(params Assembly[] assemblies) // Default to Lifestyle.Transient
ByInterface(Assembly assembly, Lifestyle lifestyle) // Set lifestyle
ByInterface(Lifestyle lifestyle, params Assembly[] assemblies) // Lifestyle is first parameter?

File: Src/Xer.Cqrs.Extensions.SimpleInjector/CqrsBuilder.cs (hunk 2 above)
Comment (jeyjeyemem) [reply Src/Xer.Cqrs.Extensions.SimpleInjector/CqrsBuilder.cs]: Oh, shouldn't private functions be camel-cased?

File: Src/Xer.Cqrs.Extensions.SimpleInjector/ICqrsEventHandlerSelector.cs (hunk 1 above)
Comment (mvput) [reply Src/Xer.Cqrs.Extensions.SimpleInjector/ICqrsEventHandlerSelector.cs]: Last solutions sound good to me pass the lifetime as the first param.

File: Src/Xer.Cqrs.Extensions.SimpleInjector/CqrsBuilder.cs (hunk 2 above)
Comment (mvput) [reply Src/Xer.Cqrs.Extensions.SimpleInjector/CqrsBuilder.cs]: I always follow this guideline https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/naming-guidelines. It has no distinction between different methods (private, internal or public etc). I am in favour of following those guidelines.

**Comment by jeyjeyemem** [comment]
Updated

**Pull request merged**